### PR TITLE
Report builder help text

### PR DESCRIPTION
**Pull request by NoahCarnahan**
Adds various help text to the report builder.
cc @snopoke

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 14 errors:

* corehq/apps/userreports/reports/builder/forms.py, line 293 - E501 line too long (162 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 295 - E501 line too long (160 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 307 - E501 line too long (215 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 321 - E501 line too long (172 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 494 - E501 line too long (169 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 742 - E501 line too long (151 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 797 - E501 line too long (163 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 814 - E501 line too long (203 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 871 - E501 line too long (277 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 881 - E501 line too long (247 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 928 - E501 line too long (276 > 115 characters)
* corehq/apps/userreports/views.py, line 202 - E501 line too long (119 > 115 characters)
* corehq/apps/userreports/views.py, line 203 - E501 line too long (190 > 115 characters)
* corehq/apps/userreports/views.py, line 204 - E501 line too long (255 > 115 characters)

File: corehq/apps/userreports/reports/builder/forms.py [hunk 1]
```diff
@@ -279,6 +289,11 @@ def __init__(self, domain, report_type, *args, **kwargs):
         self.app_source_helper = ApplicationDataSourceUIHelper()
         self.app_source_helper.bootstrap(self.domain)
         report_source_fields = self.app_source_helper.get_fields()
+        report_source_help_texts = {
+            "source_type": "Form: display data from form submissions.<br/>Case: display data from your cases. You must be using case management for this option.",
+            "application": "Which application should the data come from?",
+            "source": "For cases: choose the case type to use for this report.<br/>For forms: choose the form from your application you'd like to see data on.",
```
Comment (snopoke) [inline corehq/apps/userreports/reports/builder/forms.py]: Mark for translation

File: corehq/apps/userreports/templates/userreports/base_report_builder.html [hunk 2]
```diff
@@ -1,6 +1,12 @@
 {% extends "reports_core/base_template_new.html" %}
 {% load i18n %}
 {% load crispy_forms_tags %}
+{% load hq_shared_tags %}
+
+{% block js %}{{ block.super }}
+    <script src="{% static 'hqwebapp/js/main.js' %}"></script>
+    {# or is style/js/commcarehq.js preferred? #}
```
Comment (snopoke) [inline corehq/apps/userreports/templates/userreports/base_report_builder.html]: I think `style/js/commcarehq.js` is preferred if its compatible

**Comment by snopoke** [comment]
:white_check_mark: from me once strings have been marked for translation

**Comment by dimagimon** [comment]
There are 14 errors:

* corehq/apps/userreports/reports/builder/forms.py, line 293 - E501 line too long (165 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 295 - E501 line too long (163 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 307 - E501 line too long (218 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 321 - E501 line too long (175 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 494 - E501 line too long (169 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 742 - E501 line too long (154 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 797 - E501 line too long (166 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 814 - E501 line too long (203 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 871 - E501 line too long (277 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 881 - E501 line too long (247 > 115 characters)
* corehq/apps/userreports/reports/builder/forms.py, line 928 - E501 line too long (276 > 115 characters)
* corehq/apps/userreports/views.py, line 202 - E501 line too long (119 > 115 characters)
* corehq/apps/userreports/views.py, line 203 - E501 line too long (190 > 115 characters)
* corehq/apps/userreports/views.py, line 204 - E501 line too long (255 > 115 characters)

File: corehq/apps/userreports/templates/userreports/base_report_builder.html (hunk 2 above)
Comment (NoahCarnahan) [inline corehq/apps/userreports/templates/userreports/base_report_builder.html]: Thanks, looks like `style/js/commcarehq.js` actually wasn't compatible after all.

**Comment by NoahCarnahan** [comment]
@snopoke strings have been marked

File: corehq/apps/userreports/templates/userreports/builder_report_type_select.html [hunk 3]
```diff
@@ -28,6 +41,7 @@
                 <a href="{% url "report_builder_select_source" domain "chart" %}">
                     <button class="btn btn-primary btn-large btn-report-type">
                         {% trans 'Chart' %}
+                        <span class="hq-help-template" data-content="A bar graph or a pie chart to show data from your cases or forms. You choose the property to graph."></span>
```
Comment (snopoke) [inline corehq/apps/userreports/templates/userreports/builder_report_type_select.html]: These should also be translated with `{% trans "" %}

**Comment by snopoke** [comment]
Merged but assume you'll do a followup PR

**Comment by NoahCarnahan** [comment]
https://github.com/dimagi/commcare-hq/pull/7362

**Pull request merged**